utils/utils.go

```diff
@@ -10,10 +10,23 @@ import (
 	"regexp"
 	"strings"
 
+	"github.com/blang/semver/v4"
+	"github.com/kong/deck/cprint"
 	"github.com/kong/go-kong/kong"
 )
 
-var kongVersionRegex = regexp.MustCompile(`^\d+\.\d+`)
+var (
+	kongVersionRegex = regexp.MustCompile(`^\d+\.\d+`)
+	pathRegexPattern = regexp.MustCompile(`[^a-zA-Z0-9._~/%-]`)
+
+	Kong140Version = semver.MustParse("1.4.0")
+	Kong300Version = semver.MustParse("3.0.0")
+)
+
+// IsPathRegexLike checks if a path string contains a regex pattern.
+func IsPathRegexLike(path string) bool {
+	return pathRegexPattern.MatchString(path)
+}
```

Comment:
Heavily unit test this function.

Reply:
https://github.com/Kong/deck/pull/753/commits/f96558a816b26fbfd968b90c718e2429ac3ab7c9